src/libraries/System.Net.Quic/ref/System.Net.Quic.cs

```diff
@@ -45,7 +48,26 @@ public sealed class QuicStream : System.IO.Stream
         public override int Read(byte[] buffer, int offset, int count) => throw null;
         public override void Write(byte[] buffer, int offset, int count) => throw null;
         public long StreamId => throw null;
-        public void ShutdownRead() => throw null;
-        public void ShutdownWrite() => throw null;
+        public void AbortRead() => throw null;
+        public ValueTask WriteAsync(ReadOnlyMemory<byte> data, bool endStream, System.Threading.CancellationToken cancellationToken = default) => throw null;
+        public ValueTask ShutdownWriteCompleted(System.Threading.CancellationToken cancellationToken = default) => throw null;
+    }
+    public class QuicClientConnectionOptions
+    {
+        public SslClientAuthenticationOptions ClientAuthenticationOptions { get => throw null; set => throw null; }
+        public IPEndPoint LocalEndPoint { get => throw null; set => throw null; }
+        public IPEndPoint RemoteEndPoint { get => throw null; set => throw null; }
+        public short MaxBidirectionalStreams { get => throw null; set => throw null; }
+        public short MaxUnidirectionalStreams { get => throw null; set => throw null; }
```

Comment:
These will likely need to be `long` based on QUIC spec, unless MsQuic does something different.

Either way we generally do not expose `short` in our APIs; see e.g. `IPEndPoint.Port`. Prefer `int` with some validation.

Reply:
So I think we got a bit confused on what this limit is. The limit you are referring to (which the name implies) is https://tools.ietf.org/html/draft-ietf-quic-transport-24#section-19.11. This accounts for the total number of stream that it is allowed to make.

The limits I'm currently setting in MsQuic I believe is the total number of concurrent streams for a given connection, which is different. I'm going to rename these to be clear they are the concurrent limits.

Looking into how MsQuic exposes the MAX_STREAMS frame and limit.

Reply:
Actually, I'm not sure at all at this point. I'll update these to be a long for now though.